django/test/testcases.py

```diff
@@ -1100,11 +1100,13 @@ def skip_wrapper(*args, **kwargs):
                     raise unittest.SkipTest(reason)
                 return test_func(*args, **kwargs)
             test_item = skip_wrapper
-        else:
+        elif not getattr(test_func, '__unittest_skip__', False):
             # Assume a class is decorated
             test_item = test_func
             test_item.__unittest_skip__ = CheckCondition(condition)
-        test_item.__unittest_skip_why__ = reason
+            test_item.__unittest_skip_why__ = reason
+        else:
+            test_item = test_func
```

Comment:
I think this will need adjustments has accessing `test_func` will trigger condition evaluation which could end up performing queries at the model level.